ntoskrnl/ex/sysinfo.c

```diff
@@ -1568,30 +1557,9 @@ SSI_DEF(SystemDpcBehaviourInformation)
 /* Class 25 - Full Memory Information */
 QSI_DEF(SystemFullMemoryInformation)
 {
-    PULONG Spi = (PULONG) Buffer;
-
-    PEPROCESS TheIdleProcess;
-
-    *ReqSize = sizeof(ULONG);
-
-    if (sizeof(ULONG) != Size)
-    {
-        return STATUS_INFO_LENGTH_MISMATCH;
-    }
-
-    DPRINT("SystemFullMemoryInformation\n");
-
-    TheIdleProcess = PsIdleProcess;
-
-    DPRINT("PID: %p, KernelTime: %u PFFree: %lu PFUsed: %lu\n",
-           TheIdleProcess->UniqueProcessId,
-           TheIdleProcess->Pcb.KernelTime,
-           MiFreeSwapPages,
-           MiUsedSwapPages);
-
-    *Spi = MiMemoryConsumers[MC_USER].PagesUsed;
-
-    return STATUS_SUCCESS;
+    /* Unimplemented as per https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/ex/sysinfo/query.htm?tx=76&ts=0,1254 */
+    DPRINT1("NtSetSystemInformation - SystemFullMemoryInformation not implemented\n");
```

Comment:
Huh... are you sure? Try testing that on a **checked build** of Windows.

Reply:
Btw. this property class uses the same implementation than SystemSummaryMemoryInformation.

Reply:
This is why I leave it like this and not with the `SI_XX` macro below.

In any case, this one doesn't provide a `ULONG` but a `SYSTEM_MEMORY_INFORMATION` structure, so this can't be more wrong than it currently is.

Reply:
thanks for the info!